sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -590,7 +593,7 @@ def run_ParDo(self, transform_node):
     for ix, side_pval in enumerate(transform_node.side_inputs):
       assert isinstance(side_pval, AsSideInput)
       step_name = 'SideInput-' + self._get_unique_step_name()
-      si_label = 'side%d' % ix
+      si_label = 'side-%s-%d' % (pardo_step_name, ix)
```

Comment:
Also replace call to `_get_unique_step_name` immediately above with `pardo_step_name`?

Reply:
At least the existing behavior is that each side input gets a unique step name distinct from the ParDo's step name.  I think we should keep that for now.